Dockerfile-linux.template

```diff
@@ -307,10 +307,23 @@ RUN set -eux; \
 		"pip==$PYTHON_PIP_VERSION" \
 {{ if .setuptools then ( -}}
 		"setuptools==$PYTHON_SETUPTOOLS_VERSION" \
-{{ ) else "" end -}}
+{{ ) else ( -}}
+		--no-setuptools \
+		--no-wheel \
+{{ ) end -}}
 	; \
 	rm -f get-pip.py; \
 	\
-	pip --version
+	pip --version; \
+{{ if .setuptools then ( -}}
+	# smoke test to ensure that setuptools/wheel are included
+	python -c 'import setuptools'; \
+	python -c 'import wheel'
+{{ ) else ( -}}
+	# smoke test to ensure that setuptools/wheel are not included
+	if python -c 'import setuptools' || python -c 'import wheel'; then \
```

Comment:
```console
+ python -c import setuptools
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'setuptools'
+ python -c import wheel
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'wheel'
```
🤔 Maybe these could be `2>/dev/null`

Reply:
I guess relevant to https://github.com/docker-library/python/pull/954#discussion_r1729987390, if we made these a dedicated script we could probably do a single script that's more complex, like catching the import exception directly in Python code instead?